akka-docs/src/main/paradox/remoting-artery.md

```diff
@@ -320,16 +345,126 @@ Actor classes not included in the whitelist will not be allowed to be remote dep
 
 ## Remote Security
 
-An `ActorSystem` should not be exposed via Akka Remote (Artery) over plain Aeron/UDP to an untrusted network (e.g. internet).
-It should be protected by network security, such as a firewall. There is currently no support for encryption with Artery
-so if network security is not considered as enough protection the classic remoting with
-@ref:[TLS and mutual authentication](remoting.md#remote-tls)  should be used.
+An `ActorSystem` should not be exposed via Akka Remote (Artery) over plain Aeron/UDP or TCP to an untrusted
+network (e.g. Internet). It should be protected by network security, such as a firewall. If that is not considered
+as enough protection [TLS with mutual authentication](#remote-tls) should be enabled.
 
-Best practice is that Akka remoting nodes should only be accessible from the adjacent network.
+Best practice is that Akka remoting nodes should only be accessible from the adjacent network. Note that if TLS is
+enabled with mutual authentication there is still a risk that an attacker can gain access to a valid certificate by
+compromising any node with certificates issued by the same internal PKI tree.
 
-It is also security best practice to @ref:[disable the Java serializer](#disabling-the-java-serializer) because of
+It is also security best-practice to [disable the Java serializer](#disable-java-serializer) because of
 its multiple [known attack surfaces](https://community.hpe.com/t5/Security-Research/The-perils-of-Java-deserialization/ba-p/6838995).
 
+<a id="remote-tls"></a>
+### Configuring SSL/TLS for Akka Remoting
+
+SSL can be used as the remote transport by using the `tls-tcp` transport:
+
+```
+akka.remote.artery {
+  transport = tls-tcp
+}
+```
+
+Next the actual SSL/TLS parameters have to be configured:
+
+```
+akka.remote.artery {
+  transport = tls-tcp
+
+  ssl.config-ssl-engine {
+    key-store = "/example/path/to/mykeystore.jks"
+    trust-store = "/example/path/to/mytruststore.jks"
+
+    key-store-password = ${SSL_KEY_STORE_PASSWORD}
+    key-password = ${SSL_KEY_PASSWORD}
+    trust-store-password = ${SSL_TRUST_STORE_PASSWORD}
+
+    protocol = "TLSv1.2"
+
+    enabled-algorithms = [TLS_DHE_RSA_WITH_AES_128_GCM_SHA256]
+
+    random-number-generator = "AES128CounterSecureRNG"
+  }
+}
+```
+
+Always use [substitution from environment variables](https://github.com/lightbend/config#optional-system-or-env-variable-overrides)
+for passwords. Don't define real passwords in config files.
+
+According to [RFC 7525](https://tools.ietf.org/html/rfc7525) the recommended algorithms to use with TLS 1.2 (as of writing this document) are:
+
+ * TLS_DHE_RSA_WITH_AES_128_GCM_SHA256
```

Comment:
We might want to add a note here that the GCM cipher implementations are really slow (<20 MB/s per core) in Java 8 because they don't make use of hardware features. This should be fixed in Java 9.

20 MB/s sounds like a lot for remoting but it means that for remoting intensive TLS-secured connections might make up a significant share of CPU usage.